internal/commands/app/delete_inputs_test.go

```diff
@@ -0,0 +1,137 @@
+		Name:        "app3",
+	}
+
+	testApps := []realm.App{app1, app2, app3}
+
+	t.Run("when finding apps", func(t *testing.T) {
+		t.Run("should return a client error if present", func(t *testing.T) {
+			realmClient := mock.RealmClient{}
+			realmClient.FindAppsFn = func(filter realm.AppFilter) ([]realm.App, error) {
+				return nil, errors.New("client error")
+			}
+			inputs := deleteInputs{}
+			_, err := inputs.resolveApps(nil, realmClient)
+
+			assert.Equal(t, errors.New("client error"), err)
+		})
+
+		for _, tc := range []struct {
+			description  string
+			inputs       deleteInputs
+			apps         []realm.App
+			expectedApps []realm.App
+			expectedErr  error
+		}{
+			{
+				description: "should error with input apps and no found apps",
+				inputs:      deleteInputs{Apps: []string{"app1"}},
+				expectedErr: cli.ErrAppNotFound{},
+			},
+			{
+				description: "should return no apps with no found apps nor input apps without prompting",
+			},
+			{
+				description:  "should return found apps without prompting based on inputs provided",
+				inputs:       deleteInputs{Apps: []string{"app2", "app3"}},
+				apps:         testApps,
+				expectedApps: []realm.App{app2, app3},
+			},
+		} {
+			t.Run(tc.description, func(t *testing.T) {
+				realmClient := mock.RealmClient{}
+				realmClient.FindAppsFn = func(filter realm.AppFilter) ([]realm.App, error) {
+					return tc.apps, nil
+				}
+
+				apps, err := tc.inputs.resolveApps(nil, realmClient)
+				assert.Equal(t, tc.expectedErr, err)
+				assert.Equal(t, tc.expectedApps, apps)
+			})
+		}
+
+		t.Run("should prompt user to select apps if no input provided", func(t *testing.T) {
+			realmClient := mock.RealmClient{}
+			realmClient.FindAppsFn = func(filter realm.AppFilter) ([]realm.App, error) {
+				return testApps, nil
+			}
+
+			_, console, _, ui, consoleErr := mock.NewVT10XConsole()
+			assert.Nil(t, consoleErr)
+			defer console.Close()
+
+			doneCh := make(chan (struct{}))
+			go func() {
+				defer close(doneCh)
+
+				console.ExpectString("Select App(s)")
+				console.Send("app2")
+				console.SendLine(" ")
+				console.ExpectEOF()
+			}()
+			inputs := deleteInputs{}
+			apps, err := inputs.resolveApps(ui, realmClient)
+			assert.Nil(t, err)
+
+			console.Tty().Close() // flush the writers
+			<-doneCh              // wait for procedure to complete
+
+			assert.Equal(t, []realm.App{app2}, apps)
+		})
+
+		t.Run("should print a warning if input apps are not found", func(t *testing.T) {
+			realmClient := mock.RealmClient{}
+			realmClient.FindAppsFn = func(filter realm.AppFilter) ([]realm.App, error) {
+				return testApps, nil
+			}
+
+			_, console, _, ui, consoleErr := mock.NewVT10XConsole()
+			assert.Nil(t, consoleErr)
+			defer console.Close()
+
+			doneCh := make(chan (struct{}))
+			go func() {
+				defer close(doneCh)
+				console.ExpectString("unable to delete certain apps because they were not found: nonexistent")
+				console.ExpectEOF()
+			}()
```

Comment:
I noticed some weird behavior here that is also in some other related tests. `console.ExpectString` returns an EOF error (that we do not check) if the expected string is not found by the time we read the buffer. The weird thing is that when I check the returned string from this call, I only get the first 16(?) characters instead of the entire line. I was diving into this and couldn't quite figure out why the entire log isn't being returned by this call? Similar tests also have the same behavior where the expectString call doesn't check the string parameter properly

Reply:
Nathan ran into some output funkiness when trying to use this stuff too... I think for now until we can get to the bottom of what's going on here:
* we should use `out, ui := mock.NewUI()` for tests that require no interactivity and where we want to verify the actual command output
* we should use `_, console, _, ui, err := mock.NewV10XConsole()` where we need interactivity, don't care too much about the output just yet (would love to fix this though), and can verify our tests "pass" when they complete with their various "expects"

The unchecked errors worry me...I'm surprised the patterns out there encourage this type of usage, but I think we should eventually (future ticket) look into making sure any errors here are in fact checked and reported on